src/DataFramesMeta.jl

```diff
@@ -103,78 +186,160 @@ macro col(kw)
-                    $fun =>
-                    $dest
-                end
-            else
-                # [:x] => _f => AsTable
-                if DATAFRAMES_GEQ_22
-                    t = quote
-                        DataFramesMeta.make_source_concrete($(source)) =>
-                        $fun =>
-                        AsTable
-                    end
-                # [:x] => _f
-                else
-                    t = quote
-                        DataFramesMeta.make_source_concrete($(source)) =>
-                        $fun
-                    end
-                end
-            end
-         else
-            if kw.args[1] isa Symbol
-                # y = f(:x) becomes [:x] => _f => :y
-                dest = QuoteNode(kw.args[1])
-            elseif onearg(kw.args[1], :cols)
-                # cols(n) = f(:x) becomes [:x] => _f => n
-                dest = kw.args[1].args[2]
-            end
-            t = quote
-                DataFramesMeta.make_source_concrete($(source)) =>
-                $fun =>
-                $dest
-            end
+    if nokw == false
+        lhs = ex.args[1]
+        rhs_t = ex.args[2]
+        # if lhs is a cols(y) then the rhs gets parsed as a block
+        if onearg(lhs, :cols) && rhs_t.head === :block && length(rhs_t.args) == 2
+            rhs = rhs_t.args[2]
+        else
+            rhs = rhs_t
         end
-        return t
-    elseif onearg(kw, :cols)
-        return kw.args[2]
     else
-        throw(ArgumentError("Expressions not of the form `y = f(:x)` currently disallowed."))
+        Throw(ArgumentError("This path should not be reached"))
+    end
+
+    # y = :x
+    if lhs isa Symbol && rhs isa QuoteNode
+        source = rhs
+        fun = :identity
+        dest = QuoteNode(lhs)
+
+        return quote
+            $source => $fun => $dest
+        end
+    end
+
+    # y = cols(:x)
+    if lhs isa Symbol && onearg(rhs, :cols)
+        source = rhs.args[2]
+        fun = :identity
+        dest = QuoteNode(lhs)
+
+        return quote
+            $source => $fun => $dest
+        end
+    end
+
+    # cols(:y) = :x # rhs is NOT a
+    # QuoteNode
+    if onearg(lhs, :cols) && rhs isa QuoteNode
+        source = rhs
+        fun = :identity
+        dest = lhs.args[2]
+
+        return quote
+            $source => $fun => $dest
+        end
+    end
+
+    # cols(:y) = cols(:x)
+    if onearg(lhs, :cols) && onearg(rhs, :cols)
+        source = rhs.args[2]
+        fun = :identity
+        dest = lhs.args[2]
+
+        return quote
+            $source => $fun => $dest
+        end
+    end
+
+    # y = f(:x)
+    if lhs isa Symbol
+        source, fun = get_source_fun(rhs)
+        dest = QuoteNode(lhs)
+
+        return quote
+            $source => $fun => $dest
+        end
+    end
+
+    # y = f(cols(:x))
+    # handled above
+
+    # y = :x + 1
+    # handled above
+
+    # y = cols(:x) + 1
+    # handled above
```

Comment:
Why not mention it at the place where it's handled?

Reply:
done.